acir/src/circuit/opcodes/black_box_function_call.rs

```diff
@@ -7,138 +7,376 @@ use serde::{Deserialize, Serialize};
 
 // Note: Some functions will not use all of the witness
 // So we need to supply how many bits of the witness is needed
-#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Serialize, Deserialize)]
 pub struct FunctionInput {
     pub witness: Witness,
     pub num_bits: u32,
 }
 
 #[derive(Clone, PartialEq, Eq, Serialize, Deserialize)]
-pub struct BlackBoxFuncCall {
-    pub name: BlackBoxFunc,
-    pub inputs: Vec<FunctionInput>,
-    pub outputs: Vec<Witness>,
+pub enum BlackBoxFuncCall {
+    #[allow(clippy::upper_case_acronyms)]
+    AES {
+        inputs: Vec<FunctionInput>,
+        outputs: Vec<Witness>,
+    },
+    AND {
+        lhs: FunctionInput,
+        rhs: FunctionInput,
+        output: Witness,
+    },
+    XOR {
+        lhs: FunctionInput,
+        rhs: FunctionInput,
+        output: Witness,
+    },
+    RANGE {
+        input: FunctionInput,
+    },
+    SHA256 {
+        inputs: Vec<FunctionInput>,
+        outputs: Vec<Witness>,
+    },
+    Blake2s {
+        inputs: Vec<FunctionInput>,
+        outputs: Vec<Witness>,
+    },
+    ComputeMerkleRoot {
+        leaf: FunctionInput,
+        index: FunctionInput,
+        hash_path: Vec<FunctionInput>,
+        output: Witness,
+    },
+    SchnorrVerify {
+        public_key_x: FunctionInput,
+        public_key_y: FunctionInput,
+        signature: Vec<FunctionInput>,
+        message: Vec<FunctionInput>,
+        output: Witness,
+    },
+    Pedersen {
+        inputs: Vec<FunctionInput>,
+        outputs: Vec<Witness>,
+    },
+    // 128 here specifies that this function
+    // should have 128 bits of security
+    HashToField128Security {
+        inputs: Vec<FunctionInput>,
+        output: Witness,
+    },
+    EcdsaSecp256k1 {
+        public_key_x: Vec<FunctionInput>,
+        public_key_y: Vec<FunctionInput>,
+        signature: Vec<FunctionInput>,
+        message: Vec<FunctionInput>,
```

Comment:
```suggestion
        hashed_message: Vec<FunctionInput>,
```

We're pretty consistent in specifying this is the hashed message elsewhere so let's ensure we do it here.